src/context/AuthContext.tsx

```diff
@@ -74,8 +78,16 @@ export const AuthProvider: FC<{ children: ReactNode }> = ({ children }) => {
             })
             .catch((err) => { throw new Error(err) });
     }
-
-    const updateClient = async (user: EditUserType) => {
+    const fetchClients = async (): Promise<UserType[]> => {
+        return await makeRequest('GET', '/users')
+            .then((res) => {
+                setClients(res?.data);
+                return res?.data;
+            })
+            .catch((err) => {
+                throw new Error(err);
+            });
+    }; const updateClient = async (user: EditUserType) => {
```

Comment:
po co funkcja updateClient jest w useEffect skoro nawet sama nazwa wskazuje na to, ze jest to funkcja callback'owa? 
Tu zapewne jest Twoj problem. 
Powinienes `updateClient` wywolywac na onClick i zmieniac dodatkowo jakis state, np `const [refetch, setRefetch] = useState(false)` i wtedy bedzie sie odswiezac. Teraz to za kazdym razem jak zmieni sie jedna z wielu wartosci w dependency array oraz przy pierwszym renderze Ty jednoczesnie pobierasz clientow i wysylasz jakis request to update'u.